examples/miscellaneous/plot_roc_curve_visualization_api.py

```diff
@@ -53,3 +53,34 @@
 rfc_disp = RocCurveDisplay.from_estimator(rfc, X_test, y_test, ax=ax, alpha=0.8)
 svc_disp.plot(ax=ax, alpha=0.8)
 plt.show()
+
+# %%
+# Avoiding recomputing predictions
+# --------------------------------
+# The :meth:`sklearn.metrics.RocCurveDisplay.from_estimator` method will compute
+# the predictions using the estimator and the data. However, one can already have
+# the predictions computed. Calling
+# :meth:`sklearn.metrics.RocCurveDisplay.from_estimator` will waste time by
+# recomputing the predictions. To avoid this, one can use
+# :meth:`sklearn.metrics.RocCurveDisplay.from_predictions`.
+y_pred_svc = svc.decision_function(X_test)
+y_pred_rfc = rfc.predict_proba(X_test)
+_, ax = plt.subplots()
+rfc_disp = RocCurveDisplay.from_predictions(y_test, y_pred_rfc[:, 1], ax=ax, alpha=0.8)
+svc_disp = RocCurveDisplay.from_predictions(y_test, y_pred_svc, ax=ax, alpha=0.8)
+plt.show()
```

Comment:
The legend of this plot says ``Classifier`` for both curves. I'm not sure it comes from this PR or is already in main, but I think it can be improved (The plot above has the estimator names in the legend)